OpenRA.Mods.Cnc/Traits/PoisonedByTiberium.cs

```diff
@@ -28,13 +28,19 @@ class PoisonedByTiberiumInfo : UpgradableTraitInfo, IRulesetLoaded
 		public void RulesetLoaded(Ruleset rules, ActorInfo ai) { WeaponInfo = rules.Weapons[Weapon.ToLowerInvariant()]; }
 	}
 
-	class PoisonedByTiberium : UpgradableTrait<PoisonedByTiberiumInfo>, ITick, ISync
+	class PoisonedByTiberium : UpgradableTrait<PoisonedByTiberiumInfo>, INotifyCreated, ITick, ISync
 	{
+		ResourceLayer rl;
 		[Sync] int poisonTicks;
 
 		public PoisonedByTiberium(PoisonedByTiberiumInfo info)
 			: base(info) { }
 
+		public void Created(Actor self)
+		{
+			rl = self.World.WorldActor.Trait<ResourceLayer>();
```

Comment:
This is referencing the world actor, so you could do this in the ctor and make it readonly.